src/app/services/EloCalculationService.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core';
+import { Player } from '../jscaip/Player';
+import { Utils } from '../utils/utils';
+import { EloInfo } from '../domain/EloInfo';
+
+export interface EloInfoPair {
+    playerZero: EloInfo,
+    playerOne: EloInfo,
+}
+
+export interface EloDifferences {
+    pointChangeForZero: number,
+    pointChangeForOne: number,
+}
+
+export interface EloEntry {
+    eloInfoPair: EloInfoPair,
+    winner: 'ZERO' | 'ONE' | 'DRAW',
+}
+
+@Injectable({
+    providedIn: 'root',
+})
+export class EloCalculationService {
+
+    public static getNewElos(eloEntry: EloEntry): EloInfoPair {
```

Comment:
cette fonction peut être réduite en deux en itérant sur Player.ALL

Reply:
pas pire ! mais au pris d'un ass !